### ISSUE TITLE
SREC_INFO shows different information on windows and linux

### ISSUE THREAD

**Issue by rouhento** [sierrafoxtrot/srecord#78]
Have a MOT file produced by rx-obj-elf tool (part of Renesas GCC Toolchain). The MOT file has been created in Windows environment. When reading the information of the file with srec_info tool I get:

**In Windows:**
Format: Motorola S-Record
Header: "automation.mot"
Execution Start Address: FFE00500
Data:   FF7FFFE8 - FF7FFFF3
        FFE00500 - FFE5FC73
        FFFFFF80 - FFFFFFFF

**Linux**
Format: Motorola S-Record
Header: "automation.mot"
Execution Start Address: FFE00500
Data:   FF7FFFE8 - FF7FFFF3
        FFE00500 - FFE5FC73
        FFFFFF80 - FFFFFFFFFFFFFFFF

I'm reading the exact same MOT file both in Windows and Linux. The file is meant to be used by RX-63 microcontroller that has 2MB of flash memory. For some reason Linux doesn't understand the 32-bit end address (=0xFFFF FFFF) but gives 64-bit end address instead. The flash memory range is between 0xFFE0 0000 - 0xFFFF FFFF.

Both environments have the same version of srec_info tool:
srec_info version 1.65.0 [git hash 5844fa801c]

Any ideas why this difference?

When I reduced the flash memory address range to 1MB (0xFFE0 0000 - 0xFFEF FFFF) from the linker file, the srec_info tool was able to show the information correctly also in linux environment.

**Comment by fenugrec** [comment]
Hi do you have a (smallest) example file that exhibits this ?

**Comment by rouhento** [comment]
Hello, it's been a long time and I can't find anything right now. But I remember that it worked similarly both on Linux and Windows on master branch but v1.65 didn't work. Initially we used v1.65 because we downloaded from SourceForge.

**Comment by fenugrec** [comment]
Right, I'm thinking this may have already been fixed as a side-effect of https://github.com/sierrafoxtrot/srecord/commit/dc5d568915b0062941b46779cc369722d8ffc76f
